### PR TITLE
fix some testcases in testool related to copy table

### PR DESCRIPTION
**Pull request by lightsing**
fixes:

- callDataCopyOffset_d0_g0_v0
- calldatacopy_d4(cdc_0_neg6_ff)_g0_v0
- calldatacopy_d5(cdc_0_neg6_9)_g0_v0
- TransactionToItself_d0_g0_v0 (don't know why)

differences:
not panic in busmapping anymore:
- static_CallIdentity_1_nonzeroValue_d0_g0_v0

### CODE REVIEW

**Comment by lightsing** [comment]
\b run testool

**Comment by lightsing** [comment]
\b run testool

**Comment by lightsing** [comment]
## Example CALLDATACOPY
```
      CALLDATACOPY with: size = 0x9; data_offset = 0x20; dst_offset = 0x0

      calldata_offset = 0xf
      calldata_size = 0x10
     ┌──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┐
index│00│01│02│03│04│05│06│07│08│09│0a│0b│0c│0d│0e│0f│10│11│12│13│14│15│16│17│18│19│1a│1b│1c│1d│1e│1f│
     ├──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┤
value│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│12│34│56│78│90│ab│cd│ef│01│23│45│67│89│0a│bc│de│f0│
     └──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┘
                                                    ▲                                            ▲
                                                    │                                            │
                                             calldata_start                                calldata_ends

     ┌──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┬──┐
index│20│21│22│23│24│25│26│27│28│29│2a│2b│2c│2d│2e│2f│30│31│32│33│34│35│36│37│38│39│3a│3b│3c│3d│3e│3f│
     ├──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┼──┤
value│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│00│
     └──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┴──┘
```

### Wrong calc

src_addr_end = min(calldata_offset + calldata_size, calldata_offset + data_offset) = 0x1f
src_addr = min(calldata_offset + data_offset, src_addr_end) = 0x1f

We are copying out bound bytes at `0x1f`!!

## Others

RETURNDATACOPY is a little bit diference, trying copy non-zero length out bound would fail.

But same as CALLDATACOPY, the slice of `last_callee_memory` should cut off tail.

**Comment by lightsing** [comment]
\b run testool

**Comment by lightsing** [comment]
\b run testool

**Comment by lightsing** [comment]
\b run testool

**Comment by lightsing** [comment]
\b run testool

File: zkevm-circuits/src/evm_circuit/execution/callop.rs [hunk 1]
```diff
@@ -1581,6 +1581,48 @@ mod test {
...
+                txs[0]
+                    .from(accs[2].address)
+                    .to(accs[0].address)
+                    .gas(word!("0x0a00000000"));
+            },
+            |block, _tx| block.number(0xcafeu64),
+        )
+        .unwrap();
+
+        println!("{:?}", ctx.geth_traces[0].struct_logs);
```
Comment (lispc) [inline zkevm-circuits/src/evm_circuit/execution/callop.rs]: remove this println

**Review by lispc — APPROVED** [review]

**Pull request merged**